ocw/lib/EC2.py

```diff
@@ -141,13 +141,13 @@ def cleanup_all(self):
                     'name': img['Name'],
                     'creation_datetime':  parse(img['CreationDate']),
                     'id': img['ImageId'],
-                    })
+                })
             else:
                 logger.error("[EC2][{}] Unable to parse image name '{}'".format(
                     self.__credentials.namespace, img['Name']))
 
         for key in images:
-            images[key].sort(key=lambda x: LooseVersion(x['build']))
+            images[key].sort(key=lambda x: x['creation_datetime'])
```

Comment:
WDYT , will this solve the problem ?

Reply:
I don't like it. I would make it explicit with min-age, maxage and keep versions.
The idea of keeping the last image per Flavor was, that if you will run something manual on the latest image. you have it available.